src/test/scala/mesosphere/mesos/TaskBuilderTest.scala

```diff
@@ -1614,6 +1615,36 @@ class TaskBuilderTest extends MarathonSpec with Matchers {
     assert("1001" == env("PORT_JABBER"))
   }
 
+  // Regression COPS-2072
+  test("PortsEnvWithOnlyMappingsAndUserNetworking") {
+    val command =
+      TaskBuilder.commandInfo(
+        runSpec = AppDefinition(
+          id = runSpecId,
+          container = Some(Docker(
+            network = Some(DockerInfo.Network.USER),
+            portMappings = Seq(
+              PortMapping(containerPort = 8080, hostPort = None, servicePort = 9000, protocol = "tcp", name = Some("http")),
```

Comment:
we also want a test case that sets containerPort=0 so that the "generated container port" logic is tested

Reply:
Good catch. Done.